packages/eslint-plugin/src/rules/no-unused-expressions.ts

```diff
@@ -18,17 +21,37 @@ export default util.createRule<Options, MessageIds>({
     schema: baseRule.meta.schema,
     messages: baseRule.meta.messages,
   },
-  defaultOptions: [],
-  create(context) {
+  defaultOptions: [
+    {
+      allowShortCircuit: false,
+      allowTernary: false,
+      allowTaggedTemplates: false,
+    },
+  ],
+  create(context, options) {
     const rules = baseRule.create(context);
+    const { allowShortCircuit = false, allowTernary = false } =
+      options[0] ?? {};
+
+    function isValidExpression(node: TSESTree.Node): boolean {
+      if (allowShortCircuit && node.type === AST_NODE_TYPES.LogicalExpression) {
+        return isValidExpression(node.right);
+      }
+      if (allowTernary && node.type === AST_NODE_TYPES.ConditionalExpression) {
+        return (
+          isValidExpression(node.alternate) &&
+          isValidExpression(node.consequent)
+        );
+      }
+      return (
+        node.type === AST_NODE_TYPES.OptionalCallExpression ||
+        node.type === AST_NODE_TYPES.ImportExpression
```

Comment:
ESLint support `ImportExpression` since v7.2.0.
But typescript-eslint seems support `"eslint": "^5.0.0 || ^6.0.0 || ^7.0.0". so I didn't remove it.